packages/webview_flutter/webview_flutter_platform_interface/lib/v4/src/platform_webview_widget.dart

```diff
@@ -33,5 +36,38 @@ abstract class PlatformWebViewWidget extends PlatformInterface {
   /// Builds a new WebView.
   ///
   /// Returns a Widget tree that embeds the created web view.
-  Widget build(BuildContext context);
+  Widget build(BuildParams params);
+}
+
+/// Describes the parameters necessary for displaying the platform WebView.
+@immutable
+class BuildParams {
+  /// Constructs a [BuildParams].
+  const BuildParams(
+    this.context, {
+    required this.controller,
+    this.layoutDirection = TextDirection.ltr,
+    this.gestureRecognizers,
+  });
+
+  /// Describes the part of the user interface represented by the returned
+  /// widget.
+  final BuildContext context;
+
+  /// Controls the embedded WebView for the current platform.
+  final PlatformWebViewController controller;
+
+  /// The layout direction to use for the embedded WebView.
+  final TextDirection layoutDirection;
+
+  /// Specifies which gestures should be consumed by the web view.
+  ///
+  /// It is possible for other gesture recognizers to be competing with the web
+  /// view on pointer events, e.g if the web view is inside a [ListView] the
+  /// [ListView] will want to handle vertical drags. The web view will claim
+  /// gestures that are recognized by any of the recognizers on this list.
+  ///
+  /// When this is empty or null, the web view will only handle pointer events
```

Comment:
We should make this non-nullable with empty default so there aren't two ways of expressing the same thing.